### PR TITLE
Feature/launch by UUID

### PR DESCRIPTION
**Pull request by Shadowrs**
Like -server, but based on an Int specified by adding "uuid" into the server config json. Allows you to differentiate between multiple configs with the same Server Name

### CODE REVIEW

File: src/main/java/org/parabot/core/parsers/servers/LocalServers.java [hunk 1]
```diff
@@ -86,17 +86,24 @@ public void execute() {
                 if ((bank = object.get("bank")) != null) {
                     bankTabs = (int) bank;
                 }
+                String     uuidStr = (String) object.get("uuid"); // optional
```
Comment (JKetelaar) [inline src/main/java/org/parabot/core/parsers/servers/LocalServers.java]: What if no uuid is defined in the json

File: src/main/java/org/parabot/core/Core.java [hunk 2]
```diff
@@ -28,7 +28,8 @@
 @SuppressWarnings("Duplicates")
 public class Core {
 
-    private static boolean debug;
+    public static int quickLaunchByUuid; // used like -server, but denoted by an Int rather than the server name
+	private static boolean debug; // in debug mode, we will print more detailed error messages.
```
Comment (JKetelaar) [inline src/main/java/org/parabot/core/Core.java]: This indention seems to be off

File: src/main/java/org/parabot/core/Core.java [hunk 3]
```diff
@@ -28,7 +28,8 @@
 @SuppressWarnings("Duplicates")
 public class Core {
 
-    private static boolean debug;
+    public static int quickLaunchByUuid; // used like -server, but denoted by an Int rather than the server name
```
Comment (JKetelaar) [inline src/main/java/org/parabot/core/Core.java]: Make a getter for this

File: src/main/java/org/parabot/core/ui/ServerSelector.java [hunk 4]
```diff
@@ -23,7 +24,7 @@
 
     public ServerSelector() {
         Queue<ServerComponent> widgets = getServers();
-        if (initServer != null) {
+        if (initServer != null || Core.quickLaunchByUuid > 0) {
```
Comment (JKetelaar) [inline src/main/java/org/parabot/core/ui/ServerSelector.java]: Use a getter

File: src/main/java/org/parabot/core/parsers/servers/LocalServers.java (hunk 1 above)
Comment (Shadowrs) [reply src/main/java/org/parabot/core/parsers/servers/LocalServers.java]: In which case it is ignored. Going to change default value from 0 to -1 in next commit.

**Comment by Shadowrs** [comment]
Pending merge

File: src/main/java/org/parabot/core/ui/ServerSelector.java [hunk 5]
```diff
@@ -67,20 +68,33 @@ public static ServerSelector getInstance() {
...
-                Environment.load(widget.desc);
-                return true;
+        if (Core.getQuickLaunchByUuid() > -1) { // match the pre-requested server config uuid to quick-launch
+            for (ServerComponent widget : widgets) {
+                if (widget.desc.uuid ==  Core.getQuickLaunchByUuid()) {
+                    Environment.load(widget.desc);
+                    return true;
+                }
+            }
+            System.err.println("No server config with -uuid "+Core.getQuickLaunchByUuid()+" was found to quick launch.");
```
Comment (JKetelaar) [inline src/main/java/org/parabot/core/ui/ServerSelector.java]: Could you add spaces between `+`?

File: src/main/java/org/parabot/core/ui/ServerSelector.java [hunk 6]
```diff
@@ -67,20 +68,33 @@ public static ServerSelector getInstance() {
...
+
+        if (initServer != null) {
+            final String serverName = initServer.toLowerCase(); // match the pre-requested server name to quick-launch
+            for (ServerComponent widget : widgets) {
+                if (widget.desc.getServerName().toLowerCase().equals(serverName)) {
+                    Environment.load(widget.desc);
+                    return true;
+                }
             }
+            System.err.println("No server config with -server "+serverName+" was found to quick launch.");
```
Comment (JKetelaar) [inline src/main/java/org/parabot/core/ui/ServerSelector.java]: Could you add spaces between `+`?

**Pull request merged**